src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStep2Test.java

```diff
@@ -185,4 +189,35 @@ private TestCloudSlave(String name) throws Exception {
         }
     }
 
+    @Test public void selfNameVsLabel() throws Throwable {
+        logging.recordPackage(ExecutorStepExecution.class, Level.FINE);
+        rr.then(r -> {
+            r.jenkins.setNumExecutors(0);
+            ExtensionList.lookupSingleton(DelayX.class).active = false;
+            r.createSlave("x", null, null);
+            r.createSlave("x2", "x", null);
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("parallel b1: {node('x && !x2') {semaphore 'b1'}}, b2: {node('x2') {semaphore 'b2'}}", true));
```

Comment:
More naturally just
```suggestion
            p.setDefinition(new CpsFlowDefinition("parallel b1: {node('x') {semaphore 'b1'}}, b2: {node('x') {semaphore 'b2'}}", true));
```
but I am trying to make the test (without fix) fail more reliably.